### ISSUE TITLE
Support for Unsloth AI Models

### ISSUE THREAD

**Issue by Robertou2** [FLock-io/llm-loss-validator#77]
Can you add support for Unsloth.ai models? Unsloth makes finetuning large language models like Llama-3, Mistral, Phi-4 and Gemma 2x faster, use 70% less memory, and with no degradation in accuracy!

**Comment by Robertou2** [comment]
I found the problem is in load_model that tries to load the model with adapter weights but unsloth models are full fine-tuned models so they have to be loaded using  model = AutoModelForCausalLM.from_pretrained(  model_name_or_path, token=HF_TOKEN, **model_kwargs)
Adding these lines to load_model function in validate.py, fix the error.

# check whether it is a lora weight
if "unsloth" in model_name_or_path.lower():
        logger.info("Repo is a full fine-tuned model, loading model directly")
        model = AutoModelForCausalLM.from_pretrained(
            model_name_or_path, token=HF_TOKEN, **model_kwargs
        )
